File: app/src/main/java/com/example/java/android1/java_android_notes/service/BottomSheetDialog.java
package com.example.java.android1.java_android_notes.service;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.example.java.android1.java_android_notes.MainActivity;
import com.example.java.android1.java_android_notes.R;
import com.example.java.android1.java_android_notes.data.DataNote;
import com.example.java.android1.java_android_notes.data.DataNoteSource;
import com.example.java.android1.java_android_notes.data.DataNoteSourceFirebase;
import com.example.java.android1.java_android_notes.ui.EditNoteFragment;
import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
import com.google.android.material.button.MaterialButton;

public class BottomSheetDialog extends BottomSheetDialogFragment {

    private static final String KEY_CURRENT_INDEX = "BottomSheetDialog.index";
    private int mIndex;
    private DataNoteSource mDataNoteSource;

    public BottomSheetDialog() {

    }

    public BottomSheetDialog(int index) {
        mIndex = index;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.custom_bottom_sheet_dialog, container, false);
        if (savedInstanceState != null) {
            mIndex = savedInstanceState.getInt(KEY_CURRENT_INDEX, -1);
        }
        mDataNoteSource = DataNoteSourceFirebase.getInstance();
        DataNote dataNote = mDataNoteSource.getItem(mIndex);
        MaterialButton btnEditData = view.findViewById(R.id.dialog_btn_edit);
        MaterialButton btnRemoveData = view.findViewById(R.id.dialog_btn_remove);
        TextView tvTitleNote = view.findViewById(R.id.dialog_title);
        TextView tvDescriptionNote = view.findViewById(R.id.dialog_description);
        TextView tvDateNote = view.findViewById(R.id.dialog_date);

        tvTitleNote.setText(dataNote.getNoteName());
        tvDescriptionNote.setText(dataNote.getNoteDescription());
        tvDateNote.setText(dataNote.getNoteDate());

        btnEditData.setOnClickListener((viewClick) -> {
            Navigation navigation = new Navigation(requireActivity().getSupportFragmentManager(), (MainActivity) requireActivity());
            navigation.addFragment(EditNoteFragment.newInstance(mIndex), true, false, false);
            dismiss();
        });

        btnRemoveData.setOnClickListener((viewClick) -> {
            new DialogConfirmFragment(mIndex).show(requireActivity().getSupportFragmentManager(), "Confirm Delete Note");
            dismiss();
        });

        return view;
    }

    @Override
    public void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putInt(KEY_CURRENT_INDEX, mIndex);
    }
}

File: app/src/main/java/com/example/java/android1/java_android_notes/view/dialogs/BottomSheetDialog.kt
package com.example.java.android1.java_android_notes.view.dialogs

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.java.android1.java_android_notes.databinding.CustomBottomSheetDialogBinding
import com.example.java.android1.java_android_notes.model.DataNote
import com.google.android.material.bottomsheet.BottomSheetDialogFragment

class BottomSheetDialog(
    private var dataNote: DataNote?,
    private val onRemoveListener: ((DataNote) -> Unit)?,
) : BottomSheetDialogFragment() {

    private var _binding: CustomBottomSheetDialogBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = CustomBottomSheetDialogBinding.inflate(inflater)
        if (savedInstanceState != null) {
            dataNote = savedInstanceState.getParcelable(KEY_CURRENT_INDEX)
        }
        dataNote?.apply {
            binding.dialogTitle.text = noteTitle
            binding.dialogDescription.text = noteDescription
            binding.dialogDate.text = noteDate
        }
        binding.dialogBtnEdit.setOnClickListener {
            BottomSheetDialogEditFragment(
                dataNote = dataNote
            ).show(
                requireActivity().supportFragmentManager,
                KEY_BOTTOM_SHEET_DIALOG_EDIT_NOTE
            )
            dismiss()
        }
        binding.dialogBtnRemove.setOnClickListener {
            DialogConfirmFragment(
                dataNote = dataNote,
                onSuccessConfirmListener = onRemoveListener
            ).show(
                requireActivity().supportFragmentManager,
                KEY_DIALOG_CONFIRM_DELETE
            )
            dismiss()
        }
        return binding.root
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putParcelable(KEY_CURRENT_INDEX, dataNote)
    }

    companion object {
        private const val KEY_CURRENT_INDEX = "BottomSheetDialog.index"
        private const val KEY_BOTTOM_SHEET_DIALOG_EDIT_NOTE =
            "ListOfNotesFragment.BottomSheetDialog.EditNote"
        private const val KEY_DIALOG_CONFIRM_DELETE = "ListOfNotesFragment.Dialog.ConfirmDelete"
    }
}